node/src/p2p.rs

```diff
@@ -110,33 +125,55 @@ impl P2p {
             swarm.dial(addr)?;
         }
 
+        let (cmd_tx, cmd_rx) = flume::bounded(16);
+
         tokio::spawn(async move {
             Worker {
+                cmd_rx,
                 swarm,
                 header_sub_topic_hash: header_sub_topic.hash(),
             }
             .run()
             .await;
         });
 
-        Ok(P2p {})
+        Ok(P2p { cmd_tx })
     }
 
     pub async fn next_event(&self) -> Option<P2pEvent> {
         todo!();
     }
+
+    pub async fn network_info(&self) -> Result<NetworkInfo> {
+        let (tx, rx) = oneshot::channel();
+        self.cmd_tx
+            .send_async(P2pCmd::NetworkInfo { respond_to: tx })
+            .await
+            .map_err(|_| P2pError::WorkerDied)?;
+        rx.await
+            .map_err(|_| P2pError::CmdCancelled("NetworkInfo".to_owned()))
+    }
 }
 
 impl Worker {
     async fn run(&mut self) {
+        let mut command_stream = self.cmd_rx.clone().into_stream().fuse();
         loop {
             select! {
                 ev = self.swarm.select_next_some() => {
                     if let Err(e) = self.on_swarm_event(&ev).await {
                         warn!("Failure while handling swarm event. (error: {e}, event: {ev:?})");
                     }
                 },
-                // TODO: receive command from `P2p` and handle it
+                Some(cmd) = command_stream.next() => {
+                    match cmd {
+                        P2pCmd::NetworkInfo { respond_to } => {
+                            if respond_to.send(self.swarm.network_info()).is_err() {
+                                warn!("Couldn't send response for the network info cmd");
+                            }
+                        }
+                    }
```

Comment:
This is going to get very big at some point, I suggest this:

```
Some(cmd) = command_stream.next() => self.on_command(cmd).await
```

and then something like this:

```
async fn on_command(&mut self, cmd: P2pCmd) -> Result<()> {
    match cmd {
        P2pCmd::NetworkInfo { respond_to } => {
            // We don't handle the error because we don't care if handle got dropped
            let _ = respond_to.send(self.swarm.network_info());
        }
    }
}
```

If handling of `NetworkInfo` was bigger than 3 line I would also define `fn on_network_info_cmd()`.